hr_employee_calendar_planning/models/hr_employee.py

```diff
@@ -62,52 +92,14 @@ def _regenerate_calendar(self):
                         "auto_generate": True,
                         "name": _("Auto generated calendar for employee")
                         + " %s" % self.name,
-                        "attendance_ids": [],
+                        "attendance_ids": vals_list,
                         "two_weeks_calendar": two_weeks,
                     }
                 )
                 .id
             )
         else:
-            self.resource_calendar_id.attendance_ids.unlink()
-            self.resource_calendar_id.two_weeks_calendar = two_weeks
-        seq = 0
-        for line in self.calendar_ids:
-            for attendance_line in line.calendar_id.attendance_ids:
-                if attendance_line.display_type == "line_section":
-                    continue
-                duplicate = two_weeks and not line.calendar_id.two_weeks_calendar
-                week_odd = attendance_line.week_type == "1"
-                data = attendance_line.copy_data(
-                    {
-                        "calendar_id": self.resource_calendar_id.id,
-                        "date_from": line.date_start,
-                        "date_to": line.date_end,
-                        "week_type": "0" if duplicate else attendance_line.week_type,
-                        "sequence": seq + 26 if week_odd else 24 - seq,
-                        # To make sure sequence of odd weeks is greater than 25
-                        # and sequence of even weeks is less than 25, as in
-                        # /resource/models/resource.py#L266
-                    }
-                )[0]
-                seq += 1
-                vals_list.append((0, 0, data))
-                if duplicate:
-                    data = attendance_line.copy_data(
-                        {
-                            "calendar_id": self.resource_calendar_id.id,
-                            "date_from": line.date_start,
-                            "date_to": line.date_end,
-                            "week_type": "1",
-                            "sequence": seq + 26,
-                        },
-                    )[0]
-                    seq += 1
-                    vals_list += [(0, 0, data)]
-        if two_weeks:
-            SECTION_LINES[0][2]["sequence"] = -seq
-            vals_list = SECTION_LINES + vals_list
-        self.resource_calendar_id.attendance_ids = vals_list
+            self.resource_calendar_id.attendance_ids = vals_list
```

Comment:
Why do we need this reassignation on the else?

Reply:
Remember that now the employee's automatic calendar does not create a new one each time, for this reason, when there is already one defined and we modify the employee's calendars, the calendar lines must be modified completely.

Actually this is not something new, it is something that is already done.